book/src/message_validation.md

```diff
@@ -53,21 +53,22 @@ The following is a list of the individual semantic validation steps performed by
+|----------------|---------------------------------------------------------------------------------------------------| -------------- | -------------- | ----------------------------------------------------- |
+| `ValSem100`    | Add Proposal: Identity in proposals must be unique among proposals                                | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem101`    | Add Proposal: Signature public key in proposals must be unique among proposals                    | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem102`    | Add Proposal: HPKE init key in proposals must be unique among proposals                           | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem103`    | Add Proposal: Identity in proposals must be unique among existing group members                   | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem104`    | Add Proposal: Signature public key in proposals must be unique among existing group members       | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem105`    | Add Proposal: HPKE init key in proposals must be unique among existing group members              | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem106`    | Add Proposal: required capabilities                                                               | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem107`    | Remove Proposal: Removed member must be unique among proposals                                    | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem108`    | Remove Proposal: Removed member must be an existing group member                                  | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem109`    | Update Proposal: Identity must be unchanged between existing member and new proposal              | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem110`    | Update Proposal: HPKE init key must be unique among existing members                              | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem111`    | Update Proposal: The sender of a full Commit must not include own update proposals                | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem112`    | Update Proposal: The sender of a standalone update proposal must be of type member                | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
+| `ValSem113`    | External Add Proposal: The sender of a standalone external add proposal must be of type NewMember | ✅             | ✅ | `openmls/src/group/tests/test_proposal_validation.rs` |
```

Comment:
I think it's more complicated than that. The sender of a standalone Add proposal can either be

 - An existing group member
 - A preconfigured sender (not implemented yet)
 - A sender of type `NewMember`

So I'm wondering where this validation rule comes from and what part of the spec it implements.

Reply:
It is more a validation for **External** Add proposal only so I think in this case we can omit the `existing group member`.
Perhaps I should create a new ValSem "group" just for External proposals ?
Regarding `preconfigured` sender I preferred not to represent it yet since it's not implemented but I may be wrong

Reply:
I agree it's good to leave out preconfigured senders for now and keep the PR lean. My point was that if every kind of sender is a valid sender then the whole validity check becomes meaningless. That's why I was wondering how it relates to the spec. So the question is: what exactly is being validated here?

Reply:
Ok I get it now. At the end of the day, from a receiver POV, an external proposal is just a proposal. So there's no need for this validation. I will completely remove it